py/desispec/scripts/submit_night.py

```diff
@@ -339,8 +342,20 @@ def submit_night(night, proc_obstypes=None, z_submit_types=None, queue='realtime
                                                 check_for_outputs=check_for_outputs,
                                                 resubmit_partial_complete=resubmit_partial_complete,
                                                 z_submit_types=cur_z_submit_types,
+                                                system_name=system_name,
+                                                tilenight=tilenight)
+        if (lasttile is not None and curtile != lasttile and tilenight
+            and lasttype == 'science' and len(sciences)>0):
```

Comment:
> Please clarify in comments what each does and how they are different, and consider whether some of the if/elif/else branching logic can be updated to make this more explicit.

I've changed the logic to make it clearer; if done with science exposures for a tile and use_tilenight==True, then `submit_tilenight_and_redshifts` is called, otherwise `checkfor_and_submit_joint_job` is called, as before.